Sources/MongoSwift/BSON/AnyBSONValue.swift

```diff
@@ -2,8 +2,20 @@ import Foundation
 
 /// A struct wrapping a `BSONValue` type that allows for encoding/
 /// decoding `BSONValue`s of unknown type.  
-public struct AnyBSONValue: Codable, Equatable {
-    /// The `BSONValue` wrapped by this struct. 
+public struct AnyBSONValue: Codable, Equatable, Hashable {
+    // swiftlint:disable:next legacy_hashing
+    public var hashValue: Int {
+        if let date = self.value as? Date {
```

Comment:
can we add comments explaining why these special cases are necessary?

Reply:
Done.